library/tinkerpop/src/test/java/uk/gov/gchq/gaffer/tinkerpop/generator/GafferEntityGeneratorTest.java

```diff
@@ -46,21 +55,67 @@ public void shouldConvertGafferPopVertexToGafferEntity() {
         given(vertexFeatures.properties()).willReturn(vertexPropertyFeatures);
         given(vertexPropertyFeatures.supportsNullPropertyValues()).willReturn(true);
 
-        final String vertex = "vertex";
-        final String propValue = "property value";
-        final GafferPopVertex gafferPopVertex = new GafferPopVertex(TestGroups.ENTITY, vertex, graph);
-        gafferPopVertex.property(Cardinality.list, TestPropertyNames.STRING, propValue);
+        // Create vertexes with and without properties
+        final GafferPopVertex vertexNoProps = new GafferPopVertex(TestGroups.ENTITY, vertexNoPropsId, graph);
+        final GafferPopVertex vertexWithProps = new GafferPopVertex(TestGroups.ENTITY, vertexWithPropsId, graph);
+        vertexWithProps.property(Cardinality.list, TestPropertyNames.STRING, propValue);
 
         final GafferEntityGenerator generator = new GafferEntityGenerator();
+        // When
+        final Entity entityNoProps = generator._apply(vertexNoProps);
+        final Entity entityWithProps = generator._apply(vertexWithProps);
+
+        // Then
+        assertThat(entityNoProps.getGroup()).isEqualTo(TestGroups.ENTITY);
+        assertThat(entityNoProps.getVertex()).isEqualTo(vertexNoPropsId);
```

Comment:
Is it worth checking this has empty properties?

Reply:
Good point wouldn't harm to